checkTranslations.mjs

```diff
@@ -0,0 +1,171 @@
+
+  if (!text) {
+    return null;
+  }
+
+  return { text: text || key, container: 't', file: shortName, key };
+};
+
+async function parseFile(file) {
+  const result = [];
+  const fileContents = await promises.readFile(file, 'utf8');
+
+  const ast = parser.parse(fileContents, parserOptions);
+
+  traverse.default(ast, {
+    enter(path) {
+      if (
+        path.isCallExpression() &&
+        path.node.callee.name === 't' &&
+        path.node.arguments[0].value === 'System'
+      ) {
+        result.push(processTFunction(path, file));
+      }
+
+      if (path.isJSXElement()) {
+        const noTranslate = path.node.openingElement.attributes.find(
+          a => a.name?.name === 'no-translate'
+        );
+        if (noTranslate) {
+          path.skip();
+        }
+      }
+
+      if (path.isJSXText()) {
+        result.push(processTextNode(path, file));
+      }
+    },
+  });
+
+  return result.filter(t => t);
+}
+
+const getClient = async () => {
+  const url = process.env.DBHOST ? `mongodb://${process.env.DBHOST}/` : 'mongodb://localhost/';
+  const client = new mongodb.MongoClient(url, { useUnifiedTopology: true });
+  await client.connect();
+
+  return client;
+};
+
+const checkSystemKeys = async allTexts => {
+  const client = await getClient();
+  const db = client.db(process.env.DATABASE_NAME || 'uwazi_development');
+  const collection = db.collection('translations');
+  const [firstTranslation] = await collection.find().toArray();
+  const systemContext = firstTranslation.contexts.find(c => c.id === 'System');
+  systemContext.values.forEach(t => {});
+  const textsNotInTranslations = allTexts.filter(text => {
+    let key = text.key || text.text;
+    key = key.trim().replace(/\n\s*/g, ' ');
+    return !systemContext.values.find(t => t.key === key);
+  });
+
+  client.close();
+  return textsNotInTranslations;
+};
+
+const reportNoTranslaeElement = textsWithoutTranslateElement => {
+  if (!textsWithoutTranslateElement.length) {
+    return;
+  }
+
+  textsWithoutTranslateElement.forEach(({ text, container, file }) => {
+    process.stdout.write(`\x1b[36m ${file}\x1b[37m ${text}\x1b[31m ${container}\x1b[0m \n`);
+  });
+
+  process.stdout.write(
+    ` === Found \x1b[31m ${textsWithoutTranslateElement.length} \x1b[0m texts not wrapped in a <Translate> element === \n`
+  );
+};
+
+const reportnotInTranslations = textsNotInTranslations => {
+  if (!textsNotInTranslations.length) {
+    return;
+  }
+  textsNotInTranslations.forEach(({ text, file }) => {
+    process.stdout.write(` \x1b[36m ${file} \x1b[37m ${text}\x1b[0m \n`);
+  });
+
+  process.stdout.write(
+    ` === Found \x1b[31m ${textsNotInTranslations.length} \x1b[0m texts not in translations collection ===\n`
+  );
+};
+
+async function checkTranslations(dir) {
+  const files = await getFiles(dir);
+
+  const results = await Promise.all(files.map(file => parseFile(file)));
+  const allTexts = results.reduce((acc, curr) => acc.concat(curr), []);
```

Comment:
Maybe mistaken and extremely nitpicky, but maybe results.[flat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat)()?

Reply:
much better, changed